tests/benchmark/benchmark_gptq.py

```diff
@@ -288,46 +302,22 @@ def benchmark_memory(
 else:
     is_decoder = False
 
-act_order = None
-bits = None
-group_size = None
-kernel = None
-if args.gptq:
-    if not args.gptq_model:
-        raise ValueError("The argument --gptq-model needs to be provided when benchmarking GPTQ.")
-
-    with open(os.path.join(args.gptq_model, "quantization_config.json"), "r", encoding="utf-8") as f:
-        quantize_config_dict = json.load(f)
-
-        act_order = quantize_config_dict["desc_act"]
-        bits = quantize_config_dict["bits"]
-        group_size = quantize_config_dict["group_size"]
-
-        if not args.disable_exllama:
-            # Exllama kernel can handle both the act-order / no act-order cases.
-            kernel = "exllama"
-        elif act_order:
-            kernel = "autotogptq-cuda"
-        else:
-            kernel = "autogptq-cuda-old"
-
 load_start = time.time_ns()
 if args.gptq:
-    with init_empty_weights():
-        empty_model = AutoModelForCausalLM.from_pretrained(args.model, torch_dtype=torch.float16)
-    empty_model.tie_weights()
-    model = load_quantized_model(
-        empty_model,
-        save_folder=args.gptq_model,
-        state_dict_name="model.safetensors",
+    quantization_config = GPTQConfig(
+        bits=4, disable_exllama=args.disable_exllama, disable_exllamav2=args.disable_exllamav2
+    )
+    model = autoclass.from_pretrained(
+        args.model,
+        revision=args.revision,
+        quantization_config=quantization_config,
+        torch_dtype=torch.float16,
```

Comment:
No strong feeling about it, but this depends on Transformers, while it did not before. I don't think it matters a lot though.

Reply:
I decided to go with that as we don't need to clone any model